### PR TITLE
app onboarding: document how meshed applications can access the k8s api server

### LINKED ISSUE #133: Document how applications can access the k8s api server once part of the mesh
Document how applications can access the k8s api server once part of the mesh, using either:
1. IP range exclusions (https://docs.openservicemesh.io/docs/tasks/traffic_management/iptables_redirection/#global-outbound-ip-range-exclusions)
2. Egress policy to allow access to k8s API service (over HTTPS: https://docs.openservicemesh.io/docs/tasks/traffic_management/demos/egress_policy_demo/#https-egress)

When using 2, document that the k8s api service must not be in a namespace that OSM manages.

### PR DESCRIPTION
**Pull request by SanyaKochhar**
Also some minor updates for clarity

Partially resolves #133

### CODE REVIEW

**Review by shashankram — CHANGES_REQUESTED** [review]
@SanyaKochhar Could you please keep the existing iptables doc as is, and instead reference the iptables doc and write a few lines about accessing the k8s API server in the application onboarding guide. It's important for this note regarding API server access to be present in the application onboarding guide because the exclusions must be configured at the time of deploying applications.

File: content/docs/guides/app_onboarding/_index.md [hunk 1]
```diff
@@ -19,6 +19,20 @@ The following guide describes how to onboard a Kubernetes microservice to an OSM
     - [demo/deploy-traffic-split.sh](https://github.com/openservicemesh/osm/blob/release-v0.9/demo/deploy-traffic-split.sh)
     - [demo/deploy-traffic-target.sh](https://github.com/openservicemesh/osm/blob/release-v0.9/demo/deploy-traffic-target.sh)
 
+    If a service in the mesh needs to communicate with the Kubernetes API server, the user needs to explicitly add the Kubernetes API server's cluster IP address to the list of Global outbound IP ranges for exclusion.
```
Comment (shashankram) [inline content/docs/guides/app_onboarding/_index.md:22]:
```suggestion
    If an application in the mesh needs to communicate with the Kubernetes API server, the user needs to explicitly add the Kubernetes API server's address to the list of Global outbound IP ranges for exclusion. The IP address could be a cluster IP address or a public IP address and should be appropriately excluded for connectivity to the Kubernetes API server.
```

**Review by shashankram — APPROVED** [review]

File: content/docs/guides/app_onboarding/_index.md [hunk 2]
```diff
@@ -19,6 +19,20 @@ The following guide describes how to onboard a Kubernetes microservice to an OSM
...
+     $ kubectl get svc -n default
+     NAME         TYPE        CLUSTER-IP   EXTERNAL-IP   PORT(S)   AGE
+     kubernetes   ClusterIP   10.0.0.1     <none>        443/TCP   1d
+     ```
+    2. Add this IP to the MeshConfig so that outbound traffic to it is excluded from interception by OSM's sidecar
+    ```console
+    $ kubectl patch meshconfig osm-mesh-config -n kube-system -p '{"spec":{"traffic":{"outboundIPRangeExclusionList":["10.0.0.1/32"]}}}'  --type=merge
+    meshconfig.config.openservicemesh.io/osm-mesh-config patched
+    ```
+
```
Comment (snehachhabria) [inline content/docs/guides/app_onboarding/_index.md:35]: Can we add a note here to say the that even for access to IDMS and Azure DNS the IP address need to be excluded

Comment (shashankram) [reply content/docs/guides/app_onboarding/_index.md:35]: @snehachhabria I don't think we should be listing platform-specific instructions here.

**Review by snehachhabria — APPROVED** [review]

**Pull request merged**